packages/react/src/transactions/useFollow.ts

```diff
@@ -32,36 +32,39 @@ type FollowProfilesFlowRequest = {
   followerProfileId?: string;
 };
 
-function createFollowRequestFee(followModule: FeeFollowModuleSettingsFragment): FollowRequestFee {
+function createFollowRequestFee(
+  followModule: ChargeFollowPolicy,
+  contractAddress: string,
+): FollowRequestFee {
   return {
-    amount: erc20Amount({ from: followModule.amount }),
-    contractAddress: followModule.contractAddress,
+    amount: followModule.amount,
+    contractAddress,
     recipient: followModule.recipient,
   };
 }
 
 function createFollowProfilesFlowRequest(
-  profile: ProfileFieldsFragmentWithSupportedFollowModule,
+  profile: ProfileFieldsFragment,
   activeWallet: WalletData,
   activeProfile: ProfileFieldsFragment,
 ): FollowProfilesFlowRequest {
-  const followPolicyType = getFollowPolicyTypeFromProfileFieldsFragment(profile);
-
   const baseRequest: Pick<FollowProfilesFlowRequest, 'profileId' | 'followerAddress'> = {
     profileId: profile.id,
     followerAddress: activeWallet.address,
   };
 
+  const followPolicyType = profile.followPolicy.type;
+
   switch (followPolicyType) {
     case FollowPolicyType.CHARGE:
       invariant(
```

Comment:
It should be possible to narrow down the `profile.followPolicy` discriminated union with just the `type`. Wondering if the assignment at line 56 is needed.

In other words it seems we can get away without this invariant check. Unless I am missing something.